nodes/src/nodes/audio_tts/config_resolver.py

```
@@ -0,0 +1,141 @@
+def _merge_cfg_locked_profile_engine(logical_type: str, raw: Any, cfg: Dict[str, Any]) -> Dict[str, Any]:
+    """Ensure cfg.engine matches the selected profile's preconfig engine."""
+    profile = raw.get('profile') if raw is not None and hasattr(raw, 'get') else None
+    if (not isinstance(profile, str) or not profile.strip()) and hasattr(cfg, 'get'):
+        p = cfg.get('profile')
+        if isinstance(p, str) and p.strip():
+            profile = p.strip()
+    if not isinstance(profile, str) or not profile:
+        return cfg
+    try:
+        sdef = getServiceDefinition(logical_type)
+    except Exception:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)